dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_treemap_parents.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import {
+  EuiButton,
+  EuiFormRow,
+  EuiSpacer,
+  EuiIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  htmlIdGenerator,
+  EuiTitle,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+} from '@elastic/eui';
+import { isEmpty, uniqueId } from 'lodash';
+
+export interface ParentUnitType {
+  name: string;
+  label: string;
+  type: string;
+}
+
+export const ConfigTreemapParentFields = ({ dropdownList, selectedAxis, onSelectChange }: any) => {
+  const addButtonText = '+ Add Parent';
+
+  const options = dropdownList.map((item) => {
+    return {
+      ...item,
+      label: item.name,
+    };
+  });
+
+  const getParentUnit = () => {
```

Comment:
can just initial state for it in object and use it directly instead of having method?